tests/integration_test/action_service_test.py

```diff
@@ -50,6 +50,516 @@ def test_index():
+                                       'kairon_user_msg': None, 'session_started': None}}
+
+
+@responses.activate
+@mock.patch("kairon.shared.cloud.utils.CloudUtility.trigger_lambda", autospec=True)
+def test_pyscript_action_execution_without_pyscript_evaluator_url_raise_exception(mock_trigger_lambda):
+    import textwrap
+
+    action_name = "test_pyscript_action_execution_without_pyscript_evaluator_url_raise_exception"
+    Actions(name=action_name, type=ActionType.pyscript_action.value,
+            bot="5f50fd0a56b698ca10d35d2z", user="user").save()
+    script = """
+    numbers = [1, 2, 3, 4, 5]
+    total = 0
+    for i in numbers:
+        total += i
+    print(total)
+    """
+    script = textwrap.dedent(script)
+    PyscriptActionConfig(
+        name=action_name,
+        source_code=script,
+        bot="5f50fd0a56b698ca10d35d2z",
+        user="user"
+    ).save()
+    mock_environment = {"evaluator": {"pyscript": {"url": None}}}
+
+    request_object = {
+        "next_action": action_name,
+        "tracker": {
+            "sender_id": "default",
+            "conversation_id": "default",
+            "slots": {"bot": "5f50fd0a56b698ca10d35d2z", "location": "Bangalore", "langauge": "Kannada"},
+            "latest_message": {'text': 'get intents', 'intent_ranking': [{'name': 'pyscript_action'}]},
+            "latest_event_time": 1537645578.314389,
+            "followup_action": "action_listen",
+            "paused": False,
+            "events": [{"event1": "hello"}, {"event2": "how are you"}],
+            "latest_input_channel": "rest",
+            "active_loop": {},
+            "latest_action": {},
+        },
+        "domain": {
+            "config": {},
+            "session_config": {},
+            "intents": [],
+            "entities": [],
+            "slots": {"bot": "5f50fd0a56b698ca10d35d2z"},
+            "responses": {},
+            "actions": [],
+            "forms": {},
+            "e2e_actions": []
+        },
+        "version": "version"
+    }
+    with patch("kairon.shared.utils.Utility.environment", new=mock_environment):
+        mock_trigger_lambda.return_value = {"Payload": {"body": "Failed to evaluated the pyscript"}, "StatusCode": 422}
+        response = client.post("/webhook", json=request_object)
+        response_json = response.json()
+        assert response.status_code == 200
+        assert len(response_json['events']) == 1
+        assert len(response_json['responses']) == 1
+        assert response_json['events'] == [
+        {'event': 'slot', 'timestamp': None, 'name': 'kairon_action_response',
+         'value': "I have failed to process your request"}]
+        log = ActionServerLogs.objects(action=action_name).get().to_mongo().to_dict()
+        assert log['exception'] == "{'Payload': {'body': 'Failed to evaluated the pyscript'}, 'StatusCode': 422}"
+
+
+@responses.activate
+def test_pyscript_action_execution_with_error():
```

Comment:
kind of exception thrown is not right.
 At run time, we can expect connection error or server unreachable kind of error. It is expected that evaluation service will never throw exceptions out directly

Reply:
so, change it into connection error